source/common/upstream/upstream_impl.cc

```diff
@@ -1106,6 +1108,19 @@ ClusterInfoImpl::ClusterInfoImpl(
     idle_timeout_ = std::chrono::hours(1);
   }
 
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.tcp_pool_idle_timeout")) {
```

Comment:
doesn't applying the runtime guard here mean it won't actually reload unless you reload the cluster?  I think we may want to instead create the idle alarm based on the runtime guard such that the flag takes effect on connection creation

Reply:
You mean we'll reload the cluster only if two `ClusterInfoImpl` objects around CDS/guard_flip are not equal? Yeah happy to defer the timer decision to connection creation.